openlibrary/catalog/add_book/__init__.py

```diff
@@ -646,21 +654,35 @@ def load_data(rec, account_key=None):
         del rec['cover']
     try:
         # get an OL style edition dict
-        edition = build_query(rec)
+        rec_as_edition = build_query(rec)
+        edition: dict[str, Any]
+        if existing_edition:
+            edition = build_query(existing_edition.dict()) | rec_as_edition
```

Comment:
Maybe we can do this?

```suggestion
            edition = existing_edition.dict() | rec_as_edition
```

For `authors`, we might want to only override if the field does not exist on the existing edition. Thinking being:

- if the v1 promise item is a low quality record, it won't have authors ; problem solved
- if the v1 promise items is high quality, it _will_ have authors ; again problem solved

Reply:
I have made these changes.